course3/homework-list/benchmark.cc

```diff
@@ -0,0 +1,24 @@
+#include "list.h"
+
+#include <list>
+#include <forward_list>
+#include <algorithm>
+#include <benchmark/benchmark.h>
+
+
+template<template<typename...> class ListType>
+void ListBenchmark(benchmark::State& state)
+{
+	ListType<int> l;
+	for (int i=0; i < state.range_x(); ++i)
+		l.push_front(i);
+}
+
+//BENCHMARK_TEMPLATE(ListBenchmark, Karun::List)->Range(1<<0, 1<<10);
```

Comment:
The error is pretty clear: ```library not found for -lbenchmark```
Do you have the library installed in a place where the linker can find it? On Linux, I'd be looking for a file named ```libbenchmark.a``` or ```libbenchmark.so.X.Y``` in one of the standard system paths. I don't know what the convention is on your OS.